task/show-sbom-rhdh/0.1/show-sbom-rhdh.yaml

```diff
@@ -0,0 +1,60 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: show-sbom-rhdh
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: "docker"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "containers, rhtap"
+    task.results.format: application/text
+    task.results.key: LINK_TO_SBOM
+    task.output.location: results
+spec:
+  description: >-
+    Shows the Software Bill of Materials (SBOM) generated for the built image in CyloneDX JSON format.
+    The 'task.*' annotations are processed by Red Hat Developer Hub (RHDH) so that the log content can be rendered in its UI.
+  params:
+    - name: IMAGE_URL
+      description: Fully qualified image name to show SBOM for.
+      type: string
+  results:
+  - description: Placeholder result meant to make RHDH identify this task as the producer of the SBOM logs.
+    name: LINK_TO_SBOM
+  steps:
+  - name: annotate-task
+    image: registry.redhat.io/openshift4/ose-cli:4.13@sha256:73df37794ffff7de1101016c23dc623e4990810390ebdabcbbfa065214352c7c
+    script: |
+      #!/usr/bin/env bash
+      echo "show-sbom-rhdh $(context.taskRun.name)"
+
+      # When this task is used in a pipelineRun triggered by Pipelines as Code, the annotations will be cleared,
+      # so we're re-adding them here
+      oc annotate taskrun $(context.taskRun.name) task.results.format=application/text
+      oc annotate taskrun $(context.taskRun.name) task.results.key=LINK_TO_SBOM
+      oc annotate taskrun $(context.taskRun.name) task.output.location=logs
+  - name: show-sbom
+    image: quay.io/redhat-appstudio/cosign:v2.1.1@sha256:c883d6f8d39148f2cea71bff4622d196d89df3e510f36c140c097b932f0dd5d5
+    env:
+    - name: IMAGE_URL
+      value: $(params.IMAGE_URL)
+    script: |
+      #!/busybox/sh
+      status=-1
+      max_try=5
+      wait_sec=2
+      for run in $(seq 1 $max_try); do
+        status=0
+        cosign download sbom $IMAGE_URL 2>>err
+        status=$?
+        if [ "$status" -eq 0 ]; then
+          break
+        fi
+        sleep $wait_sec
+      done
+      if [ "$status" -ne 0 ]; then
+          echo "Failed to get SBOM after ${max_try} tries" >&2
+          cat err >&2
+      fi
```

Comment:
We don't write the LINK_TO_SBOM result, will this work? I think you need `touch $(results.LINK_TO_SBOM.path)`, otherwise the task won't have any results and I'm not sure what RHDH will think.

Reply:
I was really not sure about this, I kept the result in the Task definition because apparently it's what RHDH requires. But I see your point, maybe `touch` is the way to go.

Reply:
We definitely need the result in the Task spec, but if you don't write to it in the script the result will not appear in the TaskRun status. Not sure which one RHDH cares about

Reply:
In this case, since we do not have a external-link annotation set,  RHTAP UI will just ignore the value of LINK_TO_SBOM result and make use of `task.results.key: LINK_TO_SBOM` annotation to identify and choose this sbom task logs.

Having said that, I agree it is a good idea to have result value set here (maybe we can set to ${params.IMAGE_URL} atm? ) and  later when we have a valid URL we must replace the value of LINK_TO_SBOM result.

Reply:
We could also keep using the actual link to the SBOM OCI artifact (the link that this PR was previously handling). Let me know if you think it'd be better this way.

Reply:
Sure, you can add that but again the value will be ignored thats why I suggested IMAGE_URL since doesn't require much changes.

Reply:
I agree. I'm adding IMAGE_URL for now.